x-pack/plugins/fleet/common/services/validate_package_policy.ts

```diff
@@ -46,6 +48,11 @@ export type PackagePolicyValidationResults = {
   inputs: Record<PackagePolicyInput['type'], PackagePolicyInputValidationResults> | null;
 } & PackagePolicyConfigValidationResults;
 
+export const DATASET_REGEX = /^(?![_\.])[a-z0-9_\.]+$/;
```

Comment:
I feel like ES index name has been implemented many times throughout Kibana 😄 these example used in these two plugins are a bit more robust:
https://github.com/elastic/kibana/blob/4759173e7e98741dbc958506f2706823e12a2303/x-pack/plugins/serverless_search/public/utils/validate_index_name.ts#L10
https://github.com/elastic/kibana/blob/4759173e7e98741dbc958506f2706823e12a2303/x-pack/plugins/file_upload/public/validate_index_name.ts#L12

the only part that doesn't apply is the byte length, which should be shorter for dataset. then I remembered implementing a limit for namespace:
https://github.com/elastic/kibana/blob/e7dc1fcd035bd3ed67fd61c46809b0afa088629b/x-pack/plugins/fleet/common/services/is_valid_namespace.ts#L13

I think it would be better to create a similar separate dataset validation service rather than having it as part of this file, so that it can be more robust. WDYT?

Reply:
Thanks @jen-huang! I think we need more restriction that a valid index name (e.g. we don't want to allow `-` as it can interfere with index patterns), looking at https://github.com/elastic/kibana/blob/e7dc1fcd035bd3ed67fd61c46809b0afa088629b/x-pack/plugins/fleet/common/services/is_valid_namespace.ts#L13 seems pretty much what we need. I'll try that and test.

Reply:
Updated to reuse `isValidNamespace` with some modifications.